packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -36,15 +37,24 @@ export default class WebpackConfigGenerator {
     d('Config mode:', this.mode);
   }
 
-  resolveConfig(config: Configuration | string) {
-    if (typeof config === 'string') {
+  async resolveConfig(config: Configuration | ConfigurationFactory | string) {
+    const rawConfig = (typeof config === 'string')
       // eslint-disable-next-line import/no-dynamic-require, global-require
-      return require(path.resolve(this.projectDir, config)) as Configuration;
-    }
+      ? (require(path.resolve(this.projectDir, config)) as Configuration | ConfigurationFactory)
+      : config;
 
-    return config;
+    return processConfig(this.preprocessConfig, rawConfig);
   }
 
+  // Users can override this method in a subclass to provide custom logic or
+  // configuration parameters.
+  preprocessConfig = async (config: ConfigurationFactory): Promise<Configuration> => config(
+    {},
+    {
+      mode: this.mode,
+    },
+  )
```

Comment:
Can you explain what the purpose of this is in a bit more details, perhaps with an example or two? I'm having trouble grasping it.

Reply:
This code fills in the `argv` of the configuration - see https://webpack.js.org/configuration/configuration-types/#exporting-a-function

Out of the box, wepback is supposed to pass the `argv` with all the info from the invocation. One of those fields is `mode` - it specifies whether we're in `production` or `development`.

Adding the preprocessing target (i.e. `main`, `render`, `prerender`) can also be added here, but the right place to put it is probably `env` - the first object in the list.

That said, ideally, we'd have a way to expose `env` from the plugin configuration somehow - but I intentionally left is out in this PR so it can be properly addressed with it's own PR, discussion, and everything.

Reply:
Excellent. So how would a user of `electron-forge` modify/overwrite the `preprocessConfig` function to suit their purposes? Do you have an example of that?

Reply:
This is for later PR as I said; I have a couple of ideas, but I'd like to at least have the `mode` there first to give it out as a feature.
Tbh, it's been a while since I've been touching `electron-forge`, and the code might be quite out of date here.